update_api_key.py

```
@@ -1,157 +1,827 @@
+    if current_mode != 0o600:
+        _audit_logger().warning(
+            "Insecure permissions detected on %s (mode %o). Expected 600.",
+            path,
+            current_mode,
+        )
 
 
-def update_api_key(api_key: str, use_encryption: bool = True):
-    """
-    Update API key in MCP configuration with encryption.
+def _store_in_keychain(profile: str, api_key: str, encrypted: str) -> bool:
```

Ruff (0.13.2)

Unused function argument: `api_key`

(ARG001)